src/quest_manager/models.py

```diff
@@ -19,12 +19,52 @@
 from tags.models import TagsModelMixin
 
 
+class CategoryQuerySet(models.QuerySet):
+    """
+    Custom queryset for Category model (campaigns).
+    """
+    def with_available_quests(self):
+        """
+        Annotates and filters categories that are active and have at least one
+        quest that is visible to students and not archived.
+        Uses a correlated subquery with EXISTS for performance.
+        """
+        filtered_quests = Quest.objects.filter(
+            campaign=OuterRef('pk'),
+            visible_to_students=True,
+            archived=False
+        )
+
+        return self.filter(active=True).annotate(
```

Comment:
lots of steps here on a single line, seperate them out into their own lines of code, and then add a code comment for each explaining WHY you are doing what you're doing.

Reply:
in the end, just return the queryset without any additional manipulation on the return line.